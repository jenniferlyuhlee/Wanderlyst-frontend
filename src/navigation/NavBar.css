.NavBar{
    position: fixed;
    width: 100%;
    top: 0;
    padding: 1rem;
    background-color: white;
    display: flex;
    justify-content: space-between;
    z-index: 1000;
}
/* background-color: rgb(102, 209, 149); */

.NavBar-right{
    display: flex;
    align-items: center;
}

.NavBar-link-right{
    color: #555555;
}

.NavBar-link-left:hover, .NavBar-link-right:hover{
    color: black;
}

.NavBar-link-left{
    color: rgb(94, 184, 219);
    text-align: left;
    font-size: 1.2rem;
    font-weight: bold;
}

.NavBar-link-right{
    text-align: right;
    margin-left: 1.5rem;

}

.NavBar-link-right.active{
    font-weight: bold;
    color: black;
}
