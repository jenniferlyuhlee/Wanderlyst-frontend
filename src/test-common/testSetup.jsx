import React, { useState } from "react";
import UserContext from "../auth-user/UserContext";
import WanderlystApi from "../utils/api";
import '@testing-library/jest-dom';

const demoUser = {
  username: "testadmin", 
  firstName: "testfirst",
  lastName: "testlast",
  location: "testloc",
  bio: "testbio",
  profilePic: null,
  isAdmin: true,
};

const demoTags = [
  {id: 1, name: 'Foodie', description: 'Savor the world one bite at a time. Dive into a journey packed with delicious local cuisines, hidden culinary gems, and must-try dishes that will leave your taste buds tingling.'},
  {id: 2, name: 'Explorer', description: 'Step off the beaten path and uncover the unknown. From ancient ruins to vibrant cityscapes, let your curiosity guide you through unforgettable experiences.'}
]

const UserProvider = ({ children, 
                        currUser = demoUser, 
                        setCurrUser, 
                        tags = demoTags,
                        logout,
                        hasLikedItin,
                        toggleLike }) => {
  const [user, setUser] = useState(currUser);

  return (
    <UserContext.Provider value={{ currUser: user, 
                                  setCurrUser: setUser, 
                                  tags, logout, 
                                  hasLikedItin, toggleLike}}>
      {children}
    </UserContext.Provider>
  );
};

export { UserProvider };