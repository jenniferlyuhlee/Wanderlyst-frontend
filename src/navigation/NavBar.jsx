import React, { useContext } from "react";
import { Link, NavLink } from "react-router-dom";
import UserContext from "../auth-user/UserContext";
import "./NavBar.css"

function NavBar(){
    const { currUser, logout} = useContext(UserContext);

    return(
        <nav className = "NavBar">
            <div className = "Navbar-left">
                <NavLink className = "NavBar-link-left" to = "/">
                    Wanderlyst
                </NavLink>
                {currUser ? 
                <> 
                    <NavLink className = "NavBar-link-right" to = "/explore">
                        Explore
                    </NavLink>
                    <NavLink className = "NavBar-link-right" to = "/tags">
                        Tags
                    </NavLink>
                    <NavLink className = "NavBar-link-right" to = "/share">
                        Share
                    </NavLink>
                </>
                :
                ""
                }
            </div>
            <div className = "Navbar-right">
                {currUser? 
                <>
                    <NavLink className = "NavBar-link-right" to={`users/profile/${currUser.username}`} >
                        Profile
                    </NavLink>
                    <Link className="NavBar-link-right" 
                        to="/" 
                        onClick={logout}>Logout
                    </Link>  
                </>
                :
                <>
                    <NavLink className = "NavBar-link-right" to = "/signup">
                        Signup
                    </NavLink>
                    <NavLink className = "NavBar-link-right" to = "/login">
                        Login
                    </NavLink>
                </>
                }
            </div>
        </nav>
    )
}

export default NavBar;