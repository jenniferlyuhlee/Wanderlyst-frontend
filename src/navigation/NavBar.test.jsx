import React from "react";
import { render, fireEvent } from "@testing-library/react";
import { UserProvider } from "../test-common/testSetup";
import { MemoryRouter} from "react-router-dom";
import NavBar from "./NavBar";
import { expect, describe, test, vi} from "vitest";

describe("NavBar component: currUser", () => {
    test("renders as without crashing", () => {
        render(
            <MemoryRouter>
                <UserProvider>
                    <NavBar />
                </UserProvider>
            </MemoryRouter>)
    });
    test("matches snapshot", () => {
        const {asFragment} = render(
                        <MemoryRouter>
                            <UserProvider>
                                <NavBar/>
                            </UserProvider>
                        </MemoryRouter>)
        expect(asFragment()).toMatchSnapshot();
    });
    test("renders NavBar for anonymous users", () => {
        const {getByText} = render(
                            <MemoryRouter>
                                <UserProvider>
                                    <NavBar/>
                                </UserProvider>
                            </MemoryRouter>)
        expect(getByText("Explore")).toBeInTheDocument()
        expect(getByText("Share")).toBeInTheDocument()
        expect(getByText("Profile")).toBeInTheDocument()
    });
    test("calls logout when logout clicked", () => {
        const mockLogout = vi.fn();
        const {getByText} = render(
                <MemoryRouter>
                    <UserProvider logout = {mockLogout}>
                        <NavBar/>
                    </UserProvider>
                </MemoryRouter>)
        
        // Simulate a click on the logout button
        fireEvent.click(getByText("Logout"));

        // Assert that the mockLogout function was called
        expect(mockLogout).toHaveBeenCalled();
    })
});

describe("NavBar component: anon", () => {
    test("renders as without crashing", () => {
        render(
            <MemoryRouter>
                <UserProvider currUser = {null}>
                    <NavBar />
                </UserProvider>
            </MemoryRouter>)
    });
    test("matches snapshot", () => {
        const {asFragment} = render(
                        <MemoryRouter>
                            <UserProvider currUser = {null}>
                                <NavBar/>
                            </UserProvider>
                        </MemoryRouter>)
        expect(asFragment()).toMatchSnapshot();
    });
    test("renders NavBar for anonymous users", () => {
        const {getByText} = render(
                            <MemoryRouter>
                                <UserProvider currUser = {null}>
                                    <NavBar/>
                                </UserProvider>
                            </MemoryRouter>)
        expect(getByText("Login")).toBeInTheDocument()
        expect(getByText("Signup")).toBeInTheDocument()
    });
});