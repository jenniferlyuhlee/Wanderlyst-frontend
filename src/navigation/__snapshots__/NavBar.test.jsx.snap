// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`NavBar component: anon > matches snapshot 1`] = `
<DocumentFragment>
  <nav
    class="NavBar"
  >
    <div
      class="NavBar-left"
    >
      <a
        aria-current="page"
        class="NavBar-link-left wanderlyst active"
        href="/"
      >
        WanderLyst
      </a>
    </div>
    <div
      class="NavBar-right"
    >
      <a
        class="NavBar-link-right"
        href="/signup"
      >
        Signup
      </a>
      <a
        class="NavBar-link-right"
        href="/login"
      >
        Login
      </a>
    </div>
  </nav>
</DocumentFragment>
`;

exports[`NavBar component: currUser > matches snapshot 1`] = `
<DocumentFragment>
  <nav
    class="NavBar"
  >
    <div
      class="NavBar-left"
    >
      <a
        aria-current="page"
        class="NavBar-link-left wanderlyst active"
        href="/"
      >
        WanderLyst
      </a>
      <a
        class="NavBar-link-right"
        href="/explore"
      >
        Explore
      </a>
      <a
        class="NavBar-link-right"
        href="/tags"
      >
        Tags
      </a>
      <a
        class="NavBar-link-right"
        href="/itineraries/new"
      >
        Share
      </a>
    </div>
    <div
      class="NavBar-right"
    >
      <div
        class="dropdown NavBar-link-right"
      >
        <img
          aria-expanded="false"
          class="NavBar-profilePic dropdown-toggle"
          data-bs-toggle="dropdown"
          href="#"
          role="button"
          src="/default_profile.jpg"
        />
        <ul
          class="dropdown-menu text-center p-2"
        >
          <li>
            <a
              class="NavBar-link-right"
              href="/users/profile/testadmin"
            >
              Profile
            </a>
          </li>
          <hr
            class="my-1"
          />
          <li>
            <a
              class="NavBar-link-right"
              href="/"
            >
              Logout
            </a>
          </li>
        </ul>
      </div>
    </div>
  </nav>
</DocumentFragment>
`;
